odbc/src/api/functions.rs

```diff
@@ -1215,12 +1340,210 @@ pub extern "C" fn SQLSetStmtAttr(
+                    }
+                }
+                SqlReturn::SUCCESS
+            }
+            StatementAttribute::ParamBindOffsetPtr => {
+                stmt_handle
+                    .add_diag_info(ODBCError::Unimplemented("SQL_ATTR_PARAM_BIND_OFFSET_PTR"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::ParamBindType => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_PARAM_BIND_TYPE"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::ParamOpterationPtr => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_PARAM_OPERATION_PTR"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::ParamStatusPtr => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_PARAM_STATUS_PTR"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::ParamsProcessedPtr => {
+                stmt_handle
+                    .add_diag_info(ODBCError::Unimplemented("SQL_ATTR_PARAMS_PROCESSED_PTR"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::ParamsetSize => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_PARAMSET_SIZE"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::QueryTimeout => {
+                let mut stmt_contents = stmt.write().unwrap();
+                stmt_contents.attributes.query_timeout = value as ULen;
+                SqlReturn::SUCCESS
+            }
+            StatementAttribute::RetrieveData => match FromPrimitive::from_i32(value as i32) {
+                Some(RetrieveData::Off) => SqlReturn::SUCCESS,
+                _ => {
+                    stmt_handle
+                        .add_diag_info(ODBCError::InvalidAttrValue("SQL_ATTR_RETRIEVE_DATA"));
+                    SqlReturn::ERROR
+                }
+            },
+            StatementAttribute::RowBindOffsetPtr => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_ROW_BIND_OFFSET_PTR"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::RowBindType => {
+                let mut stmt_contents = stmt.write().unwrap();
+                stmt_contents.attributes.row_bind_type = value as ULen;
+                SqlReturn::SUCCESS
+            }
+            StatementAttribute::RowNumber => {
+                let mut stmt_contents = stmt.write().unwrap();
+                stmt_contents.attributes.row_number = value as ULen;
+                SqlReturn::SUCCESS
+            }
+            StatementAttribute::RowOperationPtr => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_ROW_OPERATION_PTR"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::RowStatusPtr => {
```

Comment:
oh, I did not remember this one and the next one.
We'll have to make sure to remember to set the values for these when sqlFetch is called.

Reply:
The SQLFetch doc has a section on the row status pointer, so we should be okay!